tools/msgenc.py

```diff
@@ -5,41 +5,51 @@
 
 import argparse, ast, re
 
-charmap = {}
-
-string_regex = re.compile(r"([\"'`])(?:[\s\S])*?(?:(?<!\\)\1)")
-
 def read_charmap(path):
-    global charmap
-
     with open(path) as infile:
         charmap = infile.read()
 
     charmap = ast.literal_eval(charmap)
-    charmap = dict([(repr(k)[1:-1],chr(v)) for k,v in charmap.items()])
+    charmap = { repr(k)[1:-1] : chr(v) for k,v in charmap.items() }
 
-def cvt_str(m):
-    string = m.group(0)
+    return charmap
 
-    for orig,char in charmap.items():
-        string = string.replace(orig, char)
+def convert_text(text, charmap):
+    def cvt_str(m):
+        string = m.group(0)
 
-    return string
+        for orig,char in charmap.items():
+            string = string.replace(orig, char)
 
-if __name__ == "__main__":
+        return string
+
+    # Naive string matcher, assumes single line strings and no comments, handles escaped quotations
+    string_regex = re.compile(r"\"((?:[^\\\"\n]|\\.)*)\"")
```

Comment:
this string probably doesn't look like what you intend it to be
for example `r"\""` evaluates to `'\\"'`, not `'"'`
see https://stackoverflow.com/questions/27467870/escape-single-quote-in-raw-string-r
(didn't see this before, weird behavior)

`r'"((?:[^\\"\n]|\.)*)"'` is what you intended to write maybe?